sky/engine/core/painting/Gradient.dart

```diff
@@ -46,7 +46,7 @@ class Gradient extends _Gradient {
   }
 
   void validateColorStops(List<Color> colors, List<double> colorStops) {
-    if (colorStops != null && colors.length != colorStops.length) {
+    if (colorStops != null && (colors == null || colors.length != colorStops.length)) {
```

Comment:
Yes, colorStops can be null when colors are specified; the implementation defaults to colorStops = [0.0, 1.0].

`if (colorStops?.length == colors?.length)`  doesn't work if only colors is non-null.